File: src/jk_5/nailed/groups/Group.java
package jk_5.nailed.groups;

import jk_5.nailed.Nailed;
import jk_5.nailed.util.EnumColor;
import net.minecraft.src.EntityPlayer;

/**
 * TODO: Edit description
 *
 * @author jk-5
 */
public abstract class Group {

    protected EnumColor nameColor;
    protected String name;

    public String getGroupID() {
        return Nailed.groupRegistry.groups.inverse().get(this);
    }

    public EnumColor getNameColor() {
        return this.nameColor;
    }

    public String getChatPrefix() {
        return "";
    }

    public String getName() {
        return this.name;
    }

    public void onPlayerJoin(EntityPlayer player) {

    }

    public void onPlayerLeave(EntityPlayer player) {

    }
}

File: src/jk_5/nailed/groups/Group.scala
package jk_5.nailed.groups

import jk_5.nailed.util.EnumColor

/**
 * No description given
 *
 * @author jk-5
 */
case class Group(private final val name: String, private final val nameColor: EnumColor) {
  @inline final def getGroupID = GroupRegistry.getGroupID(this)
  @inline def getNameColor = this.nameColor
  @inline def getChatPrefix = ""
  @inline def getName = this.name
}

class GroupPlayer extends Group("Player", EnumColor.GREY)
class GroupAdmin extends Group("Admin", EnumColor.GREY) {
  override def getChatPrefix = EnumColor.GREEN + "[Admin] " + EnumColor.RESET
}

